flows/common/common.go

```diff
@@ -210,8 +206,8 @@ func Handler(fn hdrsFunc, match matchFunc, reporter *Reporter) lwotg.FlowGenerat
 		return func(tx, rx *lwotg.FlowController) {
 			// Make the channel that is used for co-ordination between the sender and receiver.
 			ch := make(chan struct{})
-			go genFunc(tx.Stop, ch)
-			go recvFunc(rx.Stop, ch)
+			go genFunc(tx.ID, tx.Stop, ch)
```

Comment:
Probably a detail but in this current implementation we could leave packets hanging in packet channel. SInce the stop channel immediately forces a return. So basically tx and rx packets at the end of a test may not be equal. 

One option to solve this is to drain the packetCh when the stop signal is received but this could introduce other complexities. Another option would be to have the sender close the packetCh when they are done sending and receiver quit when the packet channel is closed and use contexts to interrupt the whole thing if we need to.

Reply:
This is a good point -- I guess there are two cases here for practically whether this matters:

 * in the case that the sender is checking that the exact number matches, then they probably need to specify that an exact number is sent (since the packet could be in flight somewhere else like in the veth and not yet received by the other container, or on a fibre if there was a physical connection).
 * in the case that a certain rate is expected, then the determined rx+tx rates are perhaps better to look at.

that being said, this definitely is something that can occur. I'll add a TODO to determine whether we should fix it.